### PR TITLE
Prevent disperse siblings from modifying "overdue" cards

### PR DESCRIPTION
**Pull request by user1823**
https://www.reddit.com/r/Anki/comments/19e0x2l/comment/kjapnru/

If the users actually want to schedule the cards according to the FSRS algorithm, they should use `Reschedule`. Disperse siblings should not have the same effect as reschedule.

### CODE REVIEW

**Comment by L-M-Sherlock** [comment]
If so, should we calculate the fuzz range based on stability? Why not use the card's elapsed days?

**Comment by user1823** [comment]
> If so, should we calculate the fuzz range based on stability? Why not use the card's elapsed days?

Can you elaborate?

**Comment by L-M-Sherlock** [comment]
I misunderstand the "overdue". What do you mean the "overdue"? Why we shouldn't change the due when `due > last_review + max_ivl`?

**Comment by user1823** [comment]
Basically, the due date of the card is beyond the maximum limit of the fuzz range. So, it most likely means that the user has not rescheduled their cards after updating FSRS weights (or after first switching to FSRS from SM-2).

If the user chose not to reschedule the cards, disperse siblings should not force it.

There can be other causes (e.g. bugs) of `due` being greater than the maximum limit, but disperse siblings doesn't have the responsibility of correcting this issue. The issue can be corrected by using `Reschedule` before using `Disperse`.

> I misunderstand the "overdue". 

Sorry for the terminology. It makes more sense when you consider my first commit. In that, rescheduling the cards would have made them overdue (due before today). I got the term from there and didn't change it after I made the rule more general.

**Comment by user1823** [comment]
@L-M-Sherlock, what's your opinion?

This change is very simple, and I think that it can solve https://github.com/open-spaced-repetition/fsrs4anki-helper/issues/324 too.

Expertium may be having the issue because he doesn't reschedule cards after changing the parameters. I have never experienced this issue and that's probably because I always reschedule all cards after changing the parameters.

**Comment by L-M-Sherlock** [comment]
Fine. It will be a temporary workaround until I figure out the new method to disperse siblings.

**Review by L-M-Sherlock — APPROVED** [review]

**Comment by user1823** [comment]
> It will be a temporary workaround

I don't think that it is temporary, i.e., I don't think that this should be removed at any point of time.

**Comment by L-M-Sherlock** [comment]
If so, why not just check the last revlog entry to check whether the card has been manual rescheduled?

**Comment by user1823** [comment]
> why not just check the last revlog entry to check whether the card has been manual rescheduled? 

What if the cards was rescheduled with
- Anki's built-in option
- helper add-on

Code to cover both these cases would be of several lines.

Also, how do you ensure that the card was not rescheduled?  Does the interval in the revlog need to be exactly the same as that in the upper section of the card info? What if the user had used disperse siblings in the past too? In that case, the intervals would be different (but still below the maximum limit).

In short, I think that my approach is a more efficient way of dealing with this.

> we don't know whether the user don't want to reschedule a card.

If they want to reschedule, they should press `Reschedule`. If they don't press Reschedule, we assume that they don't want to do it (most likely because of the backlog).

**Comment by L-M-Sherlock** [comment]
I get it, but there is another problem:

> If they want to reschedule, they should press `Reschedule`. If they don't press Reschedule, we assume that they don't want to do it (most likely because of the backlog).

They don't want to reschedule doesn't mean they don't want to disperse siblings. But the current method doesn't allow the card to disperse because the due range is fixed in (due, due).

**Comment by user1823** [comment]
Ok, I will make another PR.
